eth/protocols/eth/celo_peer.go

```diff
@@ -0,0 +1,33 @@
+package eth
+
+import (
+	"bytes"
+
+	"github.com/celo-org/celo-blockchain/p2p"
+	"github.com/celo-org/celo-blockchain/rlp"
+)
+
+// EncodeAndSend writes an RLP-encoded message with the given code.
+// data should encode as an RLP list.
+func (p *Peer) EncodeAndSend(msgcode uint64, data []byte) error {
+	raw, err := rlp.EncodeToBytes(data)
+	if err != nil {
+		return err
+	}
+	return p.Send(msgcode, raw)
+}
+
+// Send sends the message to this peer. Since current istanbul protocol
+// has messages encoded twice, data should be twice encoded
+// with rlp encoding.
+func (p *Peer) Send(msgcode uint64, data []byte) error {
+	// Istanbul was encoding messages before sending it to the peer,
```

Comment:
Let's maybe move this comment to the `asyncMulticast` function? This way it will be in the context where the double encoding that you mention in the comment actually happens and would make more sense.
Otherwise, looks good to me.